### ISSUE TITLE
False positive "Use of moved value" on type that implements Copy

### ISSUE THREAD

**Issue by fridge-dev** [intellij-rust/intellij-rust#10871]
## Environment

* **IntelliJ Rust plugin version:** org.rust.lang (0.4.194.5382-223)
* **Rust toolchain version:** rustc 1.72.0 (5680fa18f 2023-08-23)
* **IDE name and version:** CLion 2022.3.3 Build #CL-223.8836.42, built on March 10, 2023
Runtime version: 17.0.6+1-b653.34 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
* **Operating system:** macOS 13.5.2

## Problem description

Intellij shows a false positive "Cannot Move" error when trying to copy a primitive integer type. The code compiles and runs via IDE "Run", but the syntax highlighting incorrectly highlights an error.

<img width="612" alt="Screenshot 2023-09-25 at 10 00 25 AM" src="https://github.com/intellij-rust/intellij-rust/assets/61041270/97731d1c-cb3b-45d1-8db2-a2b35e0abdeb">

This same issue was previous resolved in #3987 , but it was suggested to create a new issue for the regression.

## Steps to reproduce

`Cargo.toml`:

```toml
[package]
name = "example"
version = "0.1.0"
edition = "2021"
```

`src/main.rs`:

```rust
fn main() {
    todo!()
}

fn squared(a: i32) -> i32 {
    a * a
}
```

**Comment by fridge-dev** [comment]
Closing issue as non-issue. Although I updated my IDE and rust plugin, I found something weird was happening where my rust plugin version was not updating and also thought it was at the newest possible version. I uninstalled and reinstalled rust plugin and it worked.
